### PR TITLE
rule `no-import-test-files`: modules that end in .js

### LINKED ISSUE #215: no-import-test-files incorrectly flags modules that end in .js
[Issuehunt badges](https://issuehunt.io/r/avajs/eslint-plugin-ava/issues/215)

Using all the default options, the following line in a test file is incorrectly flagged as violating the no-import-test-files rule:

```javascript
var highlight = require('highlight.js')
```

It appears that any module name ending in `.js` will similarly be flagged as a violation. However, the semantics for node's `require` function will _only_ look for local files if they start with a dot, so this will always be importing a module, not a test file.

> There is a $30.00 open bounty on this issue. Add more on [Issuehunt](https://issuehunt.io/r/avajs/eslint-plugin-ava/issues/215).
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/avajs/eslint-plugin-ava/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

### PR DESCRIPTION
**Pull request by Manu1400**
Fixes #215.

- [x] add new tests
- [x] use require.resolve() function

### CODE REVIEW

File: rules/no-import-test-files.js [hunk 1]
```diff
@@ -1,23 +1,35 @@
 'use strict';
-const path = require('path');
+const {resolve, relative, dirname} = require('path');
 const arrify = require('arrify');
-const pkgUp = require('pkg-up');
+const {sync} = require('pkg-up');
 const multimatch = require('multimatch');
-const util = require('../util');
+const {getAvaConfig, defaultFiles, getDocsUrl} = require('../util');
```
Comment (novemberborn) [inline rules/no-import-test-files.js]: Could you revert these changes?

**Comment by novemberborn** [comment]
@Manu1400 looks like @GMartigny also took at try at solving this. His PR has fewer changes and the tests pass, would you mind having a look? #226.

For your next PR, it helps if you put something like "Fixes #215" in the PR description (I've added it above). That way it's more discoverable from the issue discussion.

**Comment by novemberborn** [comment]
@Manu1400 just happened to notice this PR includes a fix for #213 as well (https://github.com/avajs/eslint-plugin-ava/pull/221/commits/2d487046c3ecbc8626057f40bf06f76d55edba08). Here too @GMartigny made a separate PR: #228. I like your approach better though, could you open a PR just for that change?

**Comment by sindresorhus** [comment]
Closing this in favor of #226.
